vector/src/main/java/im/vector/app/core/di/SingletonModule.kt

```diff
@@ -107,7 +110,17 @@ object VectorStaticModule {
     }
 
     @Provides
-    fun providesMatrix(context: Context): Matrix {
+    fun providesMatrixConfiguration(context: Context): MatrixConfiguration {
+        return MatrixConfiguration(
+                applicationFlavor = BuildConfig.FLAVOR_DESCRIPTION,
+                roomDisplayNameFallbackProvider = VectorRoomDisplayNameFallbackProvider(context)
```

Comment:
We could probably also provide `RoomDisplayNameFallbackProvider` as a param.

Reply:
will do! :+1:

Reply:
674aea97a8db26f63bf2c72ab68412368e1d6d2e